static/js/util/util.js

```diff
@@ -168,30 +169,58 @@ export function makeCourseProgressDisplay(course, isFirst, isLast) {
  * @param {Object} profile The user profile
  * @returns {Object} Validation errors or an empty object if no errors
  */
-export function validateProfile(profile) {
+export function validateProfile(profile, requiredFields, messages) {
+  let clone = _.cloneDeep(profile);
+  let fields = Object.entries(clone).filter(pair => _.isObject(pair[1]));
+  let nullObj = (obj) => {
+    for (let key in obj) { obj[key] = null };
+  }
+  fields.map( (pair) => {
+    for (let obj of pair[1]) {
+      nullObj(obj);
+    }
+  });
   let errors = {};
+  fields.map( pair => errors[pair[0]] = pair[1]);
 
-  let required = {
-    'first_name': "Given name",
-    'last_name': "Family name",
-    'preferred_name': "Preferred name",
-    'gender': "Gender",
-    'preferred_language': "Preferred language",
-    'city': "City",
-    'country': "Country",
-    'birth_city': 'City',
-    'birth_country': "Country",
-    'date_of_birth': "Date of birth"
-  };
+  for (let keySet of requiredFields) {
+    if (!dig(profile, keySet)) {
+      insert(errors, keySet,  `${messages[keySet.slice(-1)[0]]} is required`);
+    }
+  }
+  return errors;
+}
 
-  for (let key of Object.keys(required)) {
-    if (!profile[key]) {
-      errors[key] = `${required[key]} is required`;
+export function dig (object, keys) {
+  let obj = _.cloneDeep(object);
+  let keySet = _.cloneDeep(keys);
+  while (keySet.length > 0) {
+    let key = keySet.shift();
```

Comment:
Also, `shift()` can cause subtle performance problems since you are copying the remainder of the array each time you use it: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/shift